### PR TITLE
feat: remove unnecessary use of .resolves

### PR DESCRIPTION
**Pull request by arty-name**
## fixes KILTProtocol/ticket#2040

The expect() checks of async functions in the tests can be simplified

## How to test:

yarn test

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

**Comment by rflechtner** [comment]
Are you sure it is a good idea to change these? That changes reporting of rejections, I assume – previously `expect()` would handle rejections, with `await asyncFunction()` we just throw on reject. That is caught by jest of course but the reporting is not as nice.
```
   ● rejects in expect

    expect(received).resolves.toEqual()

    Received promise rejected instead of resolved
    Rejected to value: "the reason"

       7 |     //stuff happening
       8 |
    >  9 |     await expect(reject()).resolves.toEqual(5)
         |           ^
      10 | })
      11 |
      12 | it('rejects with await', async ()=> {

      at expect (../node_modules/expect/build/index.js:178:15)
      at Object.<anonymous> (core/src/rejectsTest.spec.js:9:11)

  ● rejects with await

    thrown: "the reason"

      10 | })
      11 |
    > 12 | it('rejects with await', async ()=> {
         | ^
      13 |
      14 |     // stuff happening
      15 |

      at Object.<anonymous> (core/src/rejectsTest.spec.js:12:1)
      at TestScheduler.scheduleTests (../node_modules/@jest/core/build/TestScheduler.js:333:13)
```

**Comment by arty-name** [comment]
Yes, I am quite confident about the changes. I am convinced that the async functions should be treated the same way as sync function, because their asynchronous nature is just an implementation detail. If a function throws or rejects unexpectedly, this is a broken test or code. How exactly jest detected this breakage is irrelevant. Worse than that, replacing the standard universal await syntax with a specific syntax for this narrow purpose makes code more confusing. I am against special-casing here. I would even prefer to have a universal syntax for the expected throw/rejection case, but this seems impossible at the moment, so we have to live with two different styles:

```js
expect(() => syncFunction()).toThrow()
await expect(asyncFunction()).rejects.toThrow()
// my preferred style:
await expect(() => asyncFunction()).toThrow()

File: packages/core/src/__integrationtests__/Blockchain.spec.ts [hunk 1]
```diff
@@ -140,7 +140,7 @@ describe('Chain returns specific errors, that we check for', () => {
       version: api.extrinsicVersion,
       tip: '0x00000000000000000000000000005678',
     })
-    expect(
+    await expect(
```
[the remark this answers is not in the document]
Comment (arty-name) [reply packages/core/src/__integrationtests__/Blockchain.spec.ts]: Figured it out and improved

**Review by tjwelde — APPROVED** [review]
LGTM

**Pull request merged**